dnf-repoconfig-daemon/logger.hpp

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef DNF_REPOCONFIG_DAEMON_LOGGER_HPP
+#define DNF_REPOCONFIG_DAEMON_LOGGER_HPP
+
+#include "../libdnf/utils/logger.hpp"
+
+class JournalLogger : public libdnf::Logger {
```

Comment:
I strongly advise against explicitly using a journal or syslog API. Just write to stderr. This makes your program (1) much easier to test in restricted build environments (`make check`), (2) run in the foreground (development), and (3) portable. The beauty of systemd/journal is that it DTRT for logging already without having to worry about it at all.